cadl-extension/src/code-model-builder.ts

```diff
@@ -1411,27 +1412,41 @@ export class CodeModelBuilder {
       return this.processChoiceSchemaForUnion(type, nonNullVariants, name);
     }
 
-    const variantsMsg = nonNullVariants
-      .map((it) => {
-        if (
-          it.type.kind === "Model" ||
-          it.type.kind === "Scalar" ||
-          it.type.kind === "Intrinsic" ||
-          it.type.kind === "Enum"
-        ) {
-          return it.type.name;
+    // TODO: name from cadl-dpg
+    const unionSchema = new OrSchema(name + "Model", this.getDoc(type), {
+      summary: this.getSummary(type),
+    });
+    unionSchema.anyOf = [];
+    nonNullVariants.forEach((it) => {
+      const variantName = this.getUnionVariantName(name, it.type);
+      const schema = this.processSchema(it.type, variantName);
+      const property = new Property(variantName, this.getDoc(type), schema, {
+        summary: this.getSummary(type),
+        required: false,
+        nullable: true,
+        readOnly: false,
+      });
+      unionSchema.anyOf.push(property);
+    });
+    return this.codeModel.schemas.add(unionSchema);
+  }
+
+  private getUnionVariantName(prefix: string, type: Type): string {
+    switch (type.kind) {
+      case "Scalar":
+        return prefix + pascalCase(type.name);
+      case "Enum":
+        return prefix + pascalCase(type.name);
+      case "Model":
+        if (isArrayModelType(this.program, type)) {
+          return this.getUnionVariantName(prefix, type.indexer.value) + "Array";
```

Comment:
This may need to be updapted according to Srikanta's comment ?https://github.com/Azure/autorest.java/issues/1891#issuecomment-1379424119
```java
private List<List<Integer>> inputListOfIntegerList; // naming conventions may change but should use List instead of Array
```

Reply:
I see.

Then `List<List<List<Integer>>> inputListOfListOfIntegerList`?

Reply:
I think so. Though a bit hard to read..

Reply:
fixed